src/vcpkg/binarycaching.cpp

```diff
@@ -2447,6 +2463,20 @@ namespace vcpkg
                     const auto& token = *args.actions_runtime_token.get();
                     m_config.read.push_back(std::make_unique<GHABinaryProvider>(zip_tool, fs, buildtrees, url, token));
                 }
+
+                if (!s.upkg_templates_to_get.empty())
```

Comment:
I think this check is redundant because the loop won't do anything if the container was empty.

Reply:
Or rather, this check needs to be moved into the if on 2427